lib/Compiler.cpp

```diff
@@ -935,6 +937,11 @@ bool GetEquivalentBuiltinsWithoutGenericPointer(llvm::Module *module,
       auto substr = [&str](size_t start, size_t end) {
         return str.substr(start, end - start);
       };
+      if (!found) {
```

Comment:
I'm confused by this. Why would the data layout be missing? Also it seems it should use whatever the current compile's target is (e.g. spir vs spir64). This seems to always be 32-bit pointers.

Reply:
You are right, this is not good, I'll rework it.

Reply:
When this module has no layout, it generates a warning message when linking it with the main one, because they are not identical.